src/lib/support/ThreadOperationalDataset.h

```diff
@@ -0,0 +1,277 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <core/CHIPCore.h>
+#include <platform/internal/DeviceNetworkInfo.h>
+
+namespace chip {
+namespace Thread {
+
+class Tlv;
```

Comment:
why the forward declaration instead of #include?

Reply:
It's not supposed to be used by external components, there's no header file for it.